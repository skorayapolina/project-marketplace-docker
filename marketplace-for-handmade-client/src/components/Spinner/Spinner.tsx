import React from "react";
import classNames from "classnames";
import style from "./style.module.scss";

const Spinner = (props) => (
  <div className={classNames(style.container, {[style.inComponent]: props.inComponent})}>
    <div className={style.spinner}>
      <div className={style.circle}>
        <span className={style.line}/>
        <span className={style.line}/>
        <span className={style.line}/>
        <span className={style.line}/>
      </div>
    </div>
  </div>
);

export default Spinner;
