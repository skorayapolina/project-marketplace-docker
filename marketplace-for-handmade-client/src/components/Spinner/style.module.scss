@import "src/styles/variables";
@import "src/styles/shared";

.container {
  @extend %center;

  position: relative;

  width: 100%;
  height: 100%;
  min-height: 100px;

  padding-top: 25vh;
}

.inComponent {
  padding-top: 140px;
}

.fullscreen {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;

  z-index: 999;

  background-color: rgba(255, 255, 255, 0.5);
}

.spinner {
  perspective: 1000px;
}

.circle,
.line {
  width: 170px;
  height: 170px;
  border: 3px solid $color-basic-medium;
  border-radius: 50%;
}

.circle {
  position: relative;
  transform-style: preserve-3d;
  animation: spinn 5s infinite linear;
}

.line {
  display: block;
  position: absolute;
  top: -5px;
  right: 0;
  bottom: 0;
  left: -5px;

  &:nth-child(1) {
    transform: rotateY(36deg);
  }
  &:nth-child(2) {
    transform: rotateY(72deg);
  }
  &:nth-child(3) {
    transform: rotateY(108deg);
  }
  &:nth-child(4) {
    transform: rotateY(144deg);
  }
}

@keyframes spinn {
  100% {
    transform: rotateY(360deg) rotateX(360deg);
  }
}
