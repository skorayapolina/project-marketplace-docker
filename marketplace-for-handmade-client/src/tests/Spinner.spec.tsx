import React from "react";
import { create } from "react-test-renderer";
import Spinner from "../components/Spinner/Spinner";

describe("Spinner component", () => {
    test("Matches the snapshot", () => {
        const spinner = create(<Spinner/>);
        expect(spinner.toJSON()).toMatchSnapshot();
    });
});
