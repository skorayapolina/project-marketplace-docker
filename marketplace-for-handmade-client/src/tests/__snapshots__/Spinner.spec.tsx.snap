// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component Matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="spinner"
  >
    <div
      className="circle"
    >
      <span
        className="line"
      />
      <span
        className="line"
      />
      <span
        className="line"
      />
      <span
        className="line"
      />
    </div>
  </div>
</div>
`;
